modules/julia/CMakeLists.txt

```diff
@@ -0,0 +1,142 @@
+endif()
+
+set(JlCxx_DIR "${JlCxx_DIR}" CACHE STRING ADVANCED)
+set(Julia_BUILD "Yes" CACHE STRING ADVANCED)
+
+
+set(the_description "The Julia bindings")
+ocv_add_module(julia
+                                 opencv_core
+                                 opencv_imgproc
+                                 opencv_imgcodecs
+                                 opencv_objdetect
+                                 opencv_highgui
+                                 opencv_features2d
+                                 opencv_videoio
+)
+
+ocv_glob_module_sources()
+ocv_module_include_directories()
+ocv_warnings_disable(CMAKE_CXX_FLAGS -Wmissing-prototypes -Wmissing-declarations)
+
+ocv_add_library(${the_module} SHARED ${OPENCV_MODULE_${the_module}_HEADERS} ${OPENCV_MODULE_${the_module}_SOURCES})
+
+if(NOT JULIA_PKG_INSTALL_PATH)
+    set(JULIA_PKG_INSTALL_PATH ${CMAKE_BINARY_DIR})
+endif()
+set(JULIA_PKG_INSTALL_PATH_HOOK ${JULIA_PKG_INSTALL_PATH} CACHE STRING "" FORCE)
+mark_as_advanced(JULIA_PKG_INSTALL_PATH_HOOK)
+
+
+message(STATUS "Installing ${the_module} bindings at ${JULIA_PKG_INSTALL_PATH}")
+
+
+install(DIRECTORY package/ DESTINATION ${JULIA_PKG_INSTALL_PATH})
+install(TARGETS ${the_module} LIBRARY DESTINATION ${JULIA_PKG_INSTALL_PATH}/OpenCV/src/lib)
+
+
+if(JULIA_PKG_INSTALL_ENV)
+    set(JULIA_PKG_EXECS "Pkg.activate(${JULIA_PKG_INSTALL_ENV});${JULIA_PKG_EXECS}")
+endif()
+
+
+set(JULIA_COMMAND "\"${Julia_EXECUTABLE}\" -e \"using Pkg; ${JULIA_PKG_EXECS}\"")
+# message(STATUS "Installing Julia bindings using ${JULIA_COMMAND}")
+
+
+install(CODE "
+set(JULIA_PKG_PATH \"${JULIA_PKG_INSTALL_PATH}/OpenCV\")
+execute_process(COMMAND \"${Julia_EXECUTABLE}\" ${CMAKE_CURRENT_LIST_DIR}/package/install_package.jl \${JULIA_PKG_PATH} OUTPUT_VARIABLE JULIA_INSTALL_OUT)
+
+
+# set(JULIA_PKG_EXECS \"Pkg.develop(PackageSpec(path=\\\\\\\"${JULIA_PKG_INSTALL_PATH}/OpenCV\\\\\\\"));\")
+
+# if(${JULIA_PKG_INSTALL_ENV})
+#     set(JULIA_PKG_EXECS \"Pkg.activate(${JULIA_PKG_INSTALL_ENV});\${JULIA_PKG_EXECS}\")
+# endif()
+# set(JULIA_COMMAND \"\\\"using Pkg; \${JULIA_PKG_EXECS}\\\"\")
+
+# message(STATUS \"Installing Julia bindings using \${JULIA_COMMAND}\")
+
+# execute_process(COMMAND \"${Julia_EXECUTABLE}\" -e \${JULIA_COMMAND} OUTPUT_VARIABLE JULIA_INSTALL_OUT)
+
+message(STATUS \"Install output: \${JULIA_INSTALL_OUT}\")
+")
+
+# set(JULIA_COMMAND \"\"${Julia_EXECUTABLE}\" -e \"using Pkg; \${JULIA_PKG_EXECS}\"\")
+
+# ocv_create_module()
+
+#ocv_add_accuracy_tests()
+#ocv_add_perf_tests()
+#ocv_add_samples()
+
+message(STATUS     ${OPENCV_MODULE_${the_module}_DEPS_TO_LINK})
+
+ocv_target_link_libraries(${the_module} PUBLIC    ${OPENCV_MODULE_${the_module}_DEPS_TO_LINK}
+                                        INTERFACE ${OPENCV_MODULE_${the_module}_DEPS_TO_LINK}
+)
+ocv_target_link_libraries(${the_module} PUBLIC    ${OPENCV_MODULE_${the_module}_DEPS_EXT}
+                                        INTERFACE ${OPENCV_MODULE_${the_module}_DEPS_EXT}
```

Comment:
Do we really need this?

Reply:
I was initially registering the module as `BINDING` but now that I'm not I think this can be removed. Should I comment it out for now? (It'll be required again in phase 3 I believe)

Reply:
Update: Yes this is needed because I'm not calling ocv_create_module().